crates/rome_js_formatter/src/js/module/import_named_clause.rs

```diff
@@ -26,19 +30,54 @@ impl FormatNode for JsImportNamedClause {
             format_elements![specifier, space_token()]
         })?;
         let from = from_token.format(formatter)?;
+
         let name = named_import.format(formatter)?;
         let assertion = assertion.format_with_or_empty(formatter, |assertion| {
             format_elements![space_token(), assertion]
         })?;
+        // only wrap the name_and_rest elements when named_specifiers is bigger than 1,
+        // else we only wrap the JsNamedImportSpecifiers. this let us match the prettier test case:
+        // ```js
+        // import {somethingSuperLongsomethingSuperLo} from 'somethingSuperLongsomethingSuperLongsomethingSuperLong'
+        // ```
+        // although this import has been exceeding the max line length, prettier still keep them all in one line
+        let need_to_wrap_group_elements = {
+            let named_import = named_import?;
+            let syntax_node = named_import.syntax();
+            match syntax_node.kind() {
+                JsSyntaxKind::JS_NAMED_IMPORT_SPECIFIERS => {
+                    let specifiers =
+                        JsNamedImportSpecifiers::unwrap_cast(syntax_node.clone()).as_fields();
+                    specifiers.specifiers.syntax().children().count() > 1 || !default.is_empty()
+                }
+                _ => true,
+            }
```

Comment:
You can directly work on the `AstNode`s here, no need to use the `SyntaxNode`

```suggestion
            match named_improt? {
                JsAnyImportName::JsNamedImportSpecifiers(named_specifiers) => {
                    named_specifiers.specifiers.len() > 1 || !default.is_empty()
                }
                _ => true,
            }
```

Also, the `AstSeparatedList` offers a `len` method.